viewer/Viewer.h

```diff
@@ -111,23 +124,58 @@ namespace Enki
 		typedef QMapIterator<const std::type_info*, const std::type_info*> ManagedObjectsAliasesMapIterator;
 		ManagedObjectsAliasesMap managedObjectsAliases;
 		
+		typedef std::pair<QString, unsigned int> viewerMessage;
+		std::list<viewerMessage> messageList;
+		QString controlError1;	// dislpayed if user try to move an object in trackball mode : action forbiden to prevent glitchies
+		QString controlError2;	// dislpayed if user try to move camera in trackball mode
+		QString controlHelp;	// dislpayed if user press F1
+
+		struct ExtendedAttributes
+		{
+			bool movableByPicking;
+
+			ExtendedAttributes():movableByPicking(false){};
+		};
+		std::map<PhysicalObject*, ExtendedAttributes> objectExtendedAttributesList;
+
 		bool mouseGrabbed;
 		QPoint mouseGrabPos;
 		double wallsHeight;
-		
-		bool doDumpFrames;
+		//! to know if camera is in trackball mode
+		bool trackballView;
 		int dumpFramesCounter;
 	
+		PhysicalObject *pointedObject, *selectedObject;
+		QVector3D pointedPoint;
+
 	public:
 		ViewerWidget(World *world, QWidget *parent = 0);
 		~ViewerWidget();
 	
+		void addManagedObjectsAlias(const std::type_info* key, const std::type_info* value);
+
+		World* getWorld();
+		QVector3D getPointedPoint();
+		PhysicalObject* getPointedObject();
+		PhysicalObject* getSelectedObject();
+		bool isTrackballActivated();
+		QString getHelpString();
+		bool isMovableByPicking(PhysicalObject* object);
+
+		void setMovableByPicking(PhysicalObject* object, bool movable);
+
+		void timerEvent(QTimerEvent * event);
```

Comment:
For `setMovableByPicking()`, we should first discuss the extended parameters.

I do not understand why the container class would like to call `timerEvent()`, as it would perform the simulation step and then update the GL view. If you want to update the GL view without doing the simulation, call `updateGL()` directly.